crates/holochain_cascade/tests/tests/get_activity.rs

```diff
@@ -1,65 +1,336 @@
+/// possible to retrieve entries from the cache.
+#[tokio::test(flavor = "multi_thread")]
+async fn fill_records_entries() {
+    holochain_trace::test_run();
+
+    let test_data = ActivityTestData::valid_chain_scenario(false);
+
+    let scenario = GetActivityTestScenario::new(test_data.clone())
+        .include_agent_activity_ops_in_dht_db()
+        .await
+        .include_agent_activity_noise_ops_in_dht_db()
+        .await
+        .include_store_entry_ops_in_cache_db()
+        .await;
+
+    let options = GetActivityOptions {
+        include_valid_activity: true,
+        include_rejected_activity: false,
+        include_full_records: true,
+        get_options: GetOptions::local(),
+        ..Default::default()
+    };
+
+    let r = scenario.query_authority(options).await.unwrap();
+
+    let expected = AgentActivityResponse {
+        agent: test_data.agent.clone(),
+        valid_activity: test_data.valid_records.clone(),
+        rejected_activity: ChainItems::NotRequested,
+        warrants: vec![],
+        status: ChainStatus::Valid(test_data.chain_head.clone()),
+        highest_observed: Some(test_data.highest_observed.clone()),
+    };
+    assert_agent_activity_responses_eq!(expected, r);
+}
+
+/// Try fetching locally and remotely, and forced remotely to check that all routes return the same
+/// result
+#[tokio::test(flavor = "multi_thread")]
+async fn fetch_routes_parity() {
+    holochain_trace::test_run();
+
+    let test_data = ActivityTestData::valid_chain_scenario(false);
+
+    let scenario = GetActivityTestScenario::new(test_data.clone())
+        .include_agent_activity_ops_in_dht_db()
+        .await
+        .include_store_entry_ops_in_dht_db()
+        .await
+        .include_agent_activity_noise_ops_in_dht_db()
+        .await
+        .include_store_entry_ops_in_cache_db()
+        .await;
+
+    let options = GetActivityOptions {
+        include_valid_activity: true,
+        include_rejected_activity: false,
+        include_full_records: true,
+        get_options: GetOptions::local(),
+        ..Default::default()
+    };
+
+    // Note that we're actually sharing one dht+cache database here so the local and remote
+    // aren't quite accurate here. But the point is to exercise the code paths and not to test
+    // that local and remote data are actually merged correctly.
+
+    let local_with_remote_authority = scenario.query_authority(options.clone()).await.unwrap();
+
+    let local_as_self_authority = scenario.query_self(options.clone()).await.unwrap();
+
+    let options = GetActivityOptions {
+        include_valid_activity: true,
+        include_rejected_activity: false,
+        include_full_records: true,
+        get_options: GetOptions::network(),
+        ..Default::default()
+    };
+
+    let remote_with_remote_authority = scenario.query_authority(options.clone()).await.unwrap();
+
+    let remote_as_self_authority = scenario.query_self(options.clone()).await.unwrap();
```

Comment:
This bit is hard to grasp.

Reply:
The idea here is to take the combinations of "am I an authority" and "should I query local or remote" and make sure that you always get the same result. Which you should, because all the data is coming from one dht+cache database stores.

Why might one check that? Because it wasn't the case when I got to the code :)

Reply:
Yes, I understand. All combinations of "am I authority or not" and "local vs. remote". It's just funny I guess because we're always expecting local results, but that's because of the implementation.